src/main/java/uk/gov/register/core/RegisterContext.java

```diff
@@ -74,6 +76,10 @@ public RegisterContext(RegisterName registerName, ConfigManager configManager, E
         this.enableDownloadResource = enableDownloadResource;
         this.authenticator = authenticator;
         this.itemValidator = new ItemValidator(registerName);
+        this.consistent = true;
+
+        this.migrate();
```

Comment:
I definitely agree that it might be better not to call `migrate()` in the constructor of `RegisterContext`. I would prefer to be explicit about when we perform a database migration (in this case, on start up of the application) rather than allowing the possibility of migration occurring every time we create a new `RegisterContext` (even though I know we currently only create these on start up). I'm less worried about the consistency check existing in the constructor, but agree that it would be nice to be explicit about these things happening on start up.

Reply:
@fjavierm I think we still need to address this comment to be able to merge this. Please can we move the call for `migrate()` and `validate()` back into `RegisterApplication`? I'd prefer to be explicit about doing calling these actions, rather than them being called every time a `RegisterContext` is created.